### DISCUSSION TITLE
[Feature Request] Next Street Name

### DISCUSSION THREAD

**Discussion by skiwarz** [osmandapp/OsmAnd#20554] [Ideas]
Please add a widget to show the next street name to turn on. While driving, it would be really nice to know what street to turn onto sooner than the app currently shows. Say for example I have 20 miles to go until I exit the freeway, knowing which exit or street to take will let us follow the signs, get into the correct lane, and in general just be more prepared than the default "In 20 miles, turn right" that is shown on the screen.
The nice part is that this is a simple feature to implement. The "Street Name" widget already automatically switches from "Current Street" to "Next Street" when you're within about 20 seconds or so. Please add a toggle to this widget to allow us to force it to display either current or next street name. Or just an additional widget that always shows the next street name.

**Comment by DmitryAlexei** [comment]
Next Street name should be available in Android Auto in the latest 4.8 release (see [commit](https://github.com/osmandapp/OsmAnd/commit/26e2066e3b7b2806b5b7a27a7b39c9a296f4c090))
